src/EnergyPlus/DataZoneEquipment.cc

```diff
@@ -382,180 +379,171 @@ void GetZoneEquipmentData(EnergyPlusData &state)
-            thisZoneEquipList.EquipTypeEnum = DataZoneEquipment::ZoneEquip::Invalid;
-            thisZoneEquipList.EquipName = "";
-            thisZoneEquipList.EquipIndex = 0;
-            thisZoneEquipList.CoolingPriority = 0;
-            thisZoneEquipList.HeatingPriority = 0;
-            thisZoneEquipList.CoolingCapacity = 0;
-            thisZoneEquipList.HeatingCapacity = 0;
-            thisZoneEquipList.SequentialCoolingFractionSchedPtr = 0;
-            thisZoneEquipList.SequentialHeatingFractionSchedPtr = 0;
-
-            for (ZoneEquipTypeNum = 1; ZoneEquipTypeNum <= thisZoneEquipList.NumOfEquipTypes; ++ZoneEquipTypeNum) {
-                const int ZoneEquipTypeIdx = ZoneEquipTypeNum - 1;
-                thisZoneEquipList.EquipType(ZoneEquipTypeNum) = AlphArray(nAlphasInExtensible * ZoneEquipTypeIdx + nAlphasBeforeExtensible + 1);
-                thisZoneEquipList.EquipName(ZoneEquipTypeNum) = AlphArray(nAlphasInExtensible * ZoneEquipTypeIdx + nAlphasBeforeExtensible + 2);
-                ValidateComponent(state,
-                                  thisZoneEquipList.EquipType(ZoneEquipTypeNum),
-                                  thisZoneEquipList.EquipName(ZoneEquipTypeNum),
-                                  IsNotOK,
-                                  CurrentModuleObject);
-                if (IsNotOK) {
-                    ShowContinueError(state, "In " + CurrentModuleObject + '=' + thisZoneEquipList.Name);
-                    state.dataZoneEquip->GetZoneEquipmentDataErrorsFound = true;
-                }
-                thisZoneEquipList.CoolingPriority(ZoneEquipTypeNum) =
-                    nint(NumArray(nNumsInExtensible * ZoneEquipTypeIdx + nNumsBeforeExtensible + 1));
-                if ((thisZoneEquipList.CoolingPriority(ZoneEquipTypeNum) < 0) ||
-                    (thisZoneEquipList.CoolingPriority(ZoneEquipTypeNum) > thisZoneEquipList.NumOfEquipTypes)) {
-                    ShowSevereError(state, std::string{RoutineName} + CurrentModuleObject + "=\"" + AlphArray(1) + "\".");
-                    ShowContinueError(state,
-                                      format("invalid {}=[{}].",
-                                             cNumericFields(nNumsInExtensible * ZoneEquipTypeIdx + nNumsBeforeExtensible + 1),
-                                             thisZoneEquipList.CoolingPriority(ZoneEquipTypeNum)));
-                    ShowContinueError(state, "equipment sequence must be > 0 and <= number of equipments in the list.");
-                    if (thisZoneEquipList.CoolingPriority(ZoneEquipTypeNum) > 0)
-                        ShowContinueError(state, format("only {} in the list.", thisZoneEquipList.NumOfEquipTypes));
-                    state.dataZoneEquip->GetZoneEquipmentDataErrorsFound = true;
-                }
+            UtilityRoutines::IsNameEmpty(state, thisZoneEquipList.Name, CurrentModuleObject, state.dataZoneEquip->GetZoneEquipmentDataErrorsFound);
 
```

Comment:
This is a required field, it can't be empty and get this far.

ZoneHVAC:EquipmentList,
  A1 , \field Name
       \required-field
       \reference ZoneEquipmentLists

Reply:
Translation, you can remove the call to `IsNameEmpty`.

Reply:
e4a4d843f0b95f06cf46e63769f1173bb212859f